ngraph/test/type_prop/top_k.cpp

```diff
@@ -75,14 +77,39 @@ TYPED_TEST_P(topk_type_prop, topk_v1_partial_ouptut)
     {
         auto k = make_shared<op::Constant>(element::i32, Shape{}, 3);
         auto topk = make_shared<TypeParam>(data, k, 1, "max", "value");
+        EXPECT_EQ(topk->get_output_shape(0), Shape({2, 3}));
         EXPECT_EQ(topk->get_output_partial_shape(0), PartialShape({2, 3}));
     }
 }
 
+TYPED_TEST_P(topk_type_prop, topk_rank_static_k_unknown)
+{
+    const int64_t axis = 1;
+    const auto data_shape = Shape{1, 10, 100};
+    const auto data = make_shared<op::Parameter>(element::f32, data_shape);
+
+    {
+        const auto k = make_shared<op::Parameter>(element::i32, PartialShape({}));
+        const auto topk = make_shared<TypeParam>(data, k, axis, "max", "value");
+
+        const PartialShape fully_dynamic_axis_shape{1, Dimension{-1}, 100};
```

Comment:
Is it a good practice to use `Dimension{-1}`? May be the right one is `Dimension::dynamic()`?

Reply:
I think you're right. Passing -1 directly means relying on the implementation details. To create a dynamic dimension the suggested factory method should be used.

Reply:
For fully dynamic dimension I agree that `Dimension::dynamic()` is probably more readable and it uses default constructor,
but in general `-1`  is a way to init dynamic dimension with extreme bounds like `Dimension(16, -1)`

Reply:
You're right, I'll use `Dimension::dynamic()` in here.